openapi/src/OpenAPI/Internal/Types.hs

```diff
@@ -262,7 +269,7 @@ newtype PathPattern = PathPattern {unPathPattern :: [PathPatternPiece]}
 
 instance FromJSON PathPattern where parseJSON = withText "String" $ either fail pure . pathPatternFromText
 instance ToJSON PathPattern where toJSON = Data.Aeson.String . pathPatternToText
-instance ToJSONKey PathPattern
+instance ToJSONKey PathPattern where toJSONKey = toJSONKeyText pathPatternToText
```

Comment:
> For example, values of type `[Text]` cannot be encoded to a string, so a Map with keys of type `[Text]` is encoded as follows:
> ```
> LBC8.putStrLn $ encode $ Map.fromList [(["foo","bar","baz" :: Text], 'a')]
> -- [[["foo","bar","baz"],"a"]]
> ```
> The default implementation of ToJSONKey chooses this method of encoding a key, using the ToJSON instance of the type.

-- [ToJSONKey docs](https://hackage.haskell.org/package/aeson-1.5.6.0/docs/Data-Aeson.html#t:ToJSONKey)

Reply:
what we want, instead, is for the text value to be used, so we need to implement the `ToJSONKey` instance explicitly